File: csharp/gmsec4/dox/field/I8Field.cs
/*
 * Copyright 2007-2020 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API {

/// <summary>
/// Specialized Field class that can be used to store a signed 8-bit value.
/// </summary>

public class I8Field : Field {

  /// <summary>Creates a I8Field object with the given value</summary>
  ///
  /// <param name="name">The name of the field</param>
  /// <param name="value">The value to store in the field</param>
  ///
  /// <exception cref="GmsecException">If the name is null or contains an empty-string</exception>
  public I8Field(string name, sbyte value) {
  }

  /// <summary>Copy constructor</summary>
  /// <param name="other">The other I8Field object to copy</param>
  public I8Field(I8Field other) {
  }

  /// <summary>Returns the value associated with the field</summary>
  ///
  /// <returns>The value associated with the field</returns>
  public sbyte GetValue() {
  }

}

}

File: nodejs/gmsec5/dox/I8Field.java
/*
 * Copyright 2007-2023 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file I8Field.java
 */

package gmsec;


/**
 * Specialized Field class that can be used to retain a signed 8-bit value.
 */
public class I8Field extends Field
{
	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data A signed 8-bit value.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name contains an empty string.
	 */
	public I8Field(name, data);


	/**
	 * Constructor.
	 *
	 * @param name      Name of the field.
	 * @param data      A signed 8-bit value.
	 * @param is_header Used to indicate if Field is a header field.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name contains an empty string.
	 */
	public I8Field(name, data, is_header);


	/**
	 * Copy constructor.
	 *
	 * @param other The other I8Field object to copy.
	 *
	 * @throws SWIG_Exception Thrown if the given I8Field is null.
	 */
	public I8Field(other);


	/**
	 * Returns the data stored within the I8Field.
	 *
	 * @return A signed 8-bit value.
	 */
	public getValue();
}
